app/components/filters/date-range-filter.ts

```diff
@@ -1,63 +1,189 @@
+  }
+
+  setInitialSelection(): void {
+    const { start, end } = this.args;
+
+    if (!start && !end) {
+      return;
+    }
+
+    let matchingPreset: Presets | undefined;
+    if (start && end) {
+      matchingPreset = this.presets.find((preset) => {
+        const presetDates = this.presetDateRanges[preset];
+
+        if (presetDates) {
+          const [presetStart, presetEnd] = presetDates;
+          return start === presetStart && end === presetEnd;
+        }
+      });
+    }
+
+    if (!matchingPreset) {
+      matchingPreset = Presets.CustomRange;
+    }
+
+    this.selectedPreset = matchingPreset;
+  }
+
+  @action handleStartDateChange(newDate: string): void {
+    console.log(newDate);
+  }
+
+  @action handleEndDateChange(newDate: string): void {
+    console.log(newDate);
+  }
+
+  @action handleSelectionChange(selectedPreset?: Presets): void {
+    this.selectedPreset = selectedPreset;
+
+    if (!selectedPreset) {
+      return this.resetDates();
+    }
+
+    if (this.isCustomRange) {
+      this.start = undefined;
+      this.end = undefined;
+    } else {
+      const presetDates = this.presetDateRanges[selectedPreset];
+
+      if (presetDates) {
+        const [presetStart, presetEnd] = presetDates;
+        this.updateDates(presetStart, presetEnd);
+      }
+    }
+  }
+
+  resetDates(): void {
+    this.updateDates(undefined, undefined);
+  }
+
+  updateDates(start?: ISODateString, end?: ISODateString): void {
     this.start = start;
     this.end = end;
     this.updateQueryParams({
-      [this.args.queryParamsA]: this.start,
-      [this.args.queryParamsB]: this.end,
+      [this.args.startQueryParam]: this.start,
+      [this.args.endQueryParam]: this.end,
     });
   }
 
-  @action
-  resetDate() {
-    this.changeValue(undefined, undefined);
-  }
-
-  @action
-  hideDate() {
-    /**
-     * There was a weird interaction with a weirder solution
-     *
-     * If you...
-     * - Selected a daterange
-     * - Applied
-     * - Opened & resetted
-     * - Applied
-     * - Opened
-     * - And then hide it (don't select anything)
-     *
-     * It would display & apply the value you've used before resetting
-     *
-     * This was because the hideAction was the same as the applyAction
-     * Not defining the hideAction makes au-date-range-picker throw an error
-     *
-     * So instead, an empty hideAction?
-     * This kinda fixed it:
-     * - Now the old value would no longer be applied
-     * - But, the old value was still displayed
-     *
-     * So how do we fix it?
-     * See the silly solution below
-     */
-    /* eslint-disable no-self-assign */
-    this.start = this.start;
-    this.end = this.end;
-    /* eslint-enable no-self-assign*/
-  }
-
-  @action
-  async dateChange(e: Event, start: string, end: string) {
-    this.changeValue(start, end);
+  /**
+   *
+   * @param params object with {queryParameterName: newValue}
+   */
+  updateQueryParams(params: {
```

Comment:
I copied this from the base class. Using inheritance to share functionality like this is a bad practice and makes things hard to follow (imo). Since it's a simple function call I don't think we need an extra abstraction here.